Jellyfin.Plugin.Bookshelf/Providers/ComicBookImageProvider.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Model.Drawing;
+using MediaBrowser.Model.Entities;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Jellyfin.Plugin.Bookshelf.Providers.ComicBook
+{
+    /// The ComicBookImageProvider tries find either a image named "cover" or,
+    /// in case that fails, just takes the first image inside the archive,
+    /// hoping that it is the cover.
+    public class ComicBookImageProvider : IDynamicImageProvider
+    {
+        private readonly ILogger<ComicBookImageProvider> _logger;
+
+        private const string CBZ_FILE_EXTENSION = ".cbz";
+
+        public string Name => "Comic book archive";
+
+        public ComicBookImageProvider(ILogger<ComicBookImageProvider> logger)
+        {
+            _logger = logger;
+        }
+
+        public async Task<DynamicImageResponse> GetImage(BaseItem item, ImageType type, CancellationToken cancellationToken)
+        {
+            //Check if the file is a .cbz file
+            var extension = Path.GetExtension(item.Path);
+            if (string.Equals(extension, CBZ_FILE_EXTENSION, StringComparison.OrdinalIgnoreCase))
+            {
+                return await LoadCover(item);
+            }
+            else
+            {
+                return new DynamicImageResponse { HasImage = false };
+            }
+        }
+
+        public IEnumerable<ImageType> GetSupportedImages(BaseItem item)
+        {
+            return new List<ImageType> { ImageType.Primary };
+        }
+
+        public bool Supports(BaseItem item)
+        {
+            return item is Book;
+        }
+
+        /// <summary>
+        /// Tries to load a cover from the .cbz archive. Returns a response
+        /// with no image if nothing is found.
+        /// </summary>
+        /// <param name="BaseItem">Item to load a cover for.</param>
+        private async Task<DynamicImageResponse> LoadCover(BaseItem item)
+        {
+
+            //The image will be loaded into memory, create stream
+            var memoryStream = new MemoryStream();
+            try
+            {
+                //Open the .cbz
+                //This should return a valid reference or throw
+                using var archive = ZipFile.OpenRead(item.Path);
+
+                //If no cover is found, throw exception to log results
+                var (cover, imageFormat) = FindCoverEntryInZip(archive) ?? throw new InvalidOperationException($"No supported cover found!");
```

Comment:
```suggestion
                var (cover, imageFormat) = FindCoverEntryInZip(archive) ?? throw new InvalidOperationException("No supported cover found!");
```

Don't use string interpolation if not passing in variable